libvast/src/sketch/bloom_filter.cpp

```diff
@@ -0,0 +1,133 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/sketch/bloom_filter.hpp"
+
+#include "vast/detail/assert.hpp"
+#include "vast/detail/worm.hpp"
+#include "vast/error.hpp"
+#include "vast/fbs/bloom_filter.hpp"
+
+#include <fmt/format.h>
+
+namespace vast {
+namespace detail {
+namespace {
+
+void add(uint64_t digest, std::span<uint64_t> bits, size_t k,
+         size_t m) noexcept {
+  for (size_t i = 0; i < k; ++i) {
```

Comment:
Good call on the assertition.

We don't require non-const lvalues to be passed by pointers in our style guide, as we generally try to use return values since C++ as much as possible. Here, I wanted to make it easy for the reader to cross-check with your canonical implementation for `worm64`. Replacing the for loop with something like this is probably more readable, though:

```cpp
  for (size_t i = 0; i < k; ++i) {
    auto [upper, lower] = wide_mul(m, digest);
    bits[upper >> 6] |= (uint64_t{1} << (upper & 63));
    lower = digest;
  }
```